src/components/PadBtn/PadBtn.tsx

```diff
@@ -2,58 +2,173 @@ import React from "react";
 import { Props } from "./types";
 import { Calc } from "./types";
 
-const PadBtn: React.FC<Props> = ({ input, calc, setCalc }) => {
+const PadBtn: React.FC<Props> = ({
+  input,
+  calc,
+  setCalc,
+  pastEquations,
+  setPastEquations,
+}) => {
   const updateCalculationState = () => {
     let newCalc = { ...calc };
+    let mainStr = newCalc.displayedString;
+    const defaultCalcValue = {
+      displayedString: "0",
+      sign: [""],
+      nums: [""],
+      result: 0,
+    };
 
     if (input === "AC") {
-      setCalc({
-        displayedString: "0",
-        sign: [""],
-        nums: [""],
-        result: 0,
-      });
+      setCalc(defaultCalcValue);
+      setPastEquations([]);
+      return;
+    }
+
+    if (input === "C") {
+      setCalc(defaultCalcValue);
       return;
     }
 
     if (input === "=") {
+      if (newCalc.result === 0) newCalc.result = parseFloat(mainStr);
+      setPastEquations([
+        ...pastEquations,
+        <div key={pastEquations.length} className="history">
+          <p>{mainStr}</p>
+          <p>= {newCalc.result}</p>
+        </div>,
+      ]);
+
+      if (newCalc.result !== 0) mainStr = newCalc.result.toString();
+      newCalc.result = 0;
+      newCalc.displayedString = mainStr;
+      setCalc(newCalc);
       return;
     }
 
     if (input === "<-") {
-      newCalc.displayedString = newCalc.displayedString.slice(0, -1);
+      mainStr = mainStr.slice(0, -1);
+      if (mainStr === "") mainStr = "0";
+      newCalc.displayedString = mainStr;
       getCalculation(newCalc);
       setCalc(newCalc);
       return;
     }
 
-    if (newCalc.displayedString.charAt(0) === "0")
-      newCalc.displayedString = newCalc.displayedString.slice(0, -1); //clear first zero
+    if (input === ",") input = ".";
+
+    if (input === "xⁿ") input = "^";
 
-      let isLessThan8 = newCalc.nums[newCalc.nums.length - 1].length < 8;
-      if (isLessThan8) newCalc.displayedString += input; //need improvement
+    if (input === "±" && mainStr === "0") return;
+
+    let lastDigit = mainStr.charAt(mainStr.length - 1);
+
+    if (!isNumber(lastDigit) && !isNumber(input))
+      //if (input !== "π" && input !== "√" && lastDigit !== "π" && lastDigit !== "√")
+      mainStr = mainStr.slice(0, -1);
+
+    if (mainStr.slice(-1) === "0" && isNumber(input))
+      mainStr = mainStr.slice(0, -1); //clear first zero
+
+    let numberLenght = 0;
+    for (let i = 0; i < mainStr.length; i++) {
+      if (isNumber(mainStr.charAt(i))) numberLenght++;
+      else numberLenght = 0;
+    }
 
+    if (numberLenght < 8 || !isNumber(input)) mainStr += input;
+
+    if (input === "±") {
+      let numberToInvert = newCalc.nums[newCalc.nums.length - 1];
+      mainStr = mainStr.slice(0, -1);
+      if (newCalc.sign[newCalc.sign.length - 1] === "-") {
+        const changeToPositive = mainStr.replace(/-([^-]*)$/, "+" + "$1");
+        mainStr = changeToPositive;
```

Comment:
[suggestion] Nice, now I get what this is doing! You could also make `changeToPositive` be only the regex, like `const changeToPositive = /-([^-]*)$/, "+" + "$1";`, this way the name would make morse sense, because it literally will change the value to positive, haha.

Reply:
I can make it like `replaceNegativeSign = /-([^-]*)$/` and then `mainStr.replace(replaceNegativeSign, "+" + "$1");`. I have no clue how to make it receive all of this information, unless I create an object for it. But I will also need to create its type, so Typescript can work properly. So, how should I proceed?

Reply:
Oh sorry, my bad! You can do that, just do `mainStr.replace(replaceNegativeSign, "+" + "$1");` and it should be good.